### ISSUE TITLE
Peak detection trouble

### ISSUE THREAD

**Issue by ShusakuASANO** [Tarskin/HappyTools#53]
Hi,
I'm trying to detect peaks in my GPC (gel permeation chromatography) data.
However, no peak was detected in HappyTools. Does anyone know the reason or the proper setting?
Screenshot and data are attached.
Thanks!

![GPC chromatogram](https://user-images.githubusercontent.com/58682370/72987780-6b6dd800-3e2e-11ea-9da3-c129f3d62f6d.png)
[GPC chromatogram.txt](https://github.com/Tarskin/HappyTools/files/4103239/GPC.chromatogram.txt)

**Comment by Tarskin** [comment]
Is the intent to separate the overlapping 3 (4?) peaks in the 10-25 minute range?

**Comment by ShusakuASANO** [comment]
Yes, probabry there are 3 peaks. At least, I want to separate into 2 around 18 min and 22 min.

**Comment by Tarskin** [comment]
This is an interesting measurement, as the implemented algorithm is a bit too greedy in determining breakpoints in the f'(x) function that is applied through the top of the major peak, stating that the time-points belonging to a single f(x) range from 18.558 to 18.582. I expect that this is because the data in those maxima is not as smooth as I would expect, I am still thinking on how to avoid the program to crash (which is what happens under the hood) but there is a relatively 'simple' workaround which involves smoothing the chromatogram twice and then performing the peak detection (with your parameters, except I left number of datapoints at 100), yielding the following annotation that indicates 3 majpr peaks in that region (green, red and purple) and several smaller peaks of which brown seems plausible (as purple is trailing differently from the rest).

![image](https://user-images.githubusercontent.com/4366212/76844483-a17f7500-683d-11ea-807c-757afd8b1a0a.png)

**Comment by ShusakuASANO** [comment]
Thank you for your kind reply and suggestion.
I could reproduce the peak detection as you posted, however, the peaks are different in pdf report as the attached picture.
I did "Peak Detection" then "Save Annotation", and conducted "Quantify chromatogram" using the saved annotation.
![peaks in pdf report](https://user-images.githubusercontent.com/58682370/76860315-29956700-689e-11ea-896b-08950b3df435.png)

Is the image correct as the quantified chromatogram? I'm confused with the gap between beautiful detected peaks and the pdf reports.

Thank you,

**Comment by Tarskin** [comment]
The PDF report's first slide shows the area under the trace; determined using the default quantitation method (which yields a more robust quantitation for low abundant analytes):

![image](https://user-images.githubusercontent.com/4366212/76860932-12ce2080-685c-11ea-9a21-57387ea2809f.png)

Alternatively, the method to get the quantified areas as shown in the peak detection method would be to rely on the 'Gaussian Intensities' that can also be determined (but is currently not shown in the 'overview' slide of the PDF report).

**Comment by Tarskin** [comment]
There is a possibility to include both in the PDF, one corresponding to either method of quantitation, would this be useful for you?

**Comment by ShusakuASANO** [comment]
Thank you for the information.
I checked pdf and results file for the "Gaussian Area" and "Peak Area".
Peak #5, which was clearly detected in "Peak Detection" resulted in negative value in "Peak Area" and zero value in the "Gaussian Area".

I think that there is a mismatch between my chromatogram and the HappyTools algorithm.
Could you check the attached data and give some comments?

Thanks!
 
[GPC chromatogram.pdf](https://github.com/Tarskin/HappyTools/files/4348934/GPC.chromatogram.pdf)


Peak Area (Background Subtracted)	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	296.5999390638495	1037.361396418066	2489.855875395111	1104.2913947821303	-1166.3445486992275	-2102.8603125536215	157.58604834359957	143.31796535396757	48.788284520311144	215.7270414207622

Gaussian Area (Background Subtracted)	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	-3.858025010572419e-13	-7.349676423018536e-14	2138.728752793353	3539.2318467793543	9.851675031313789e-12	-2165.6487152754307	-250.62455012694065	237.16611885617743	221.87670968164613	199.22317872768525

Peak Noise (Standard deviation of integration window)	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	116.69142901303182	449.2689069236874	684.0835902537236	560.3136929171383	237.46537890180485	410.2562589972835	67.34796029373877	20.467876648305133	7.7910189773050975	70.8324178596875

Background	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	69.29143155086108	204.6078296926224	648.8153753130615	1063.5816547636848	555.1238910129622	200.55487706573498	126.36553177077988	82.87964290816939	119.3849769825844	49.9414287028729

Noise	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	0.9863923702708274	2.362447971293854	8.747691686860202	4.349101054665511	5.890260375782651	0.9803251854658355	0.5195883055811534	0.19998673819305535	0.0185065588102879	1.7917957424317146

GPQ (Gaussian Peak Quality)	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	0.0	0.0	0.8678702341385529	1.0	0.0	1.0	0.0	1.0	1.0	1.0

FWHM	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	0.0	0.0	2.4701108496452866	1.6377049642796873	0.0	3.2639258958838973	0.0	2.0372714290789085	2.0052400127955243	0.9823947405255229

Signal-to-Noise	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	431.4867066877723	667.4581488495845	230.62658098210596	393.9710387097246	149.72908580431823	1233.6927469889981	465.68782172509805	325.8054094411633	1436.4920114583904	130.22184800558884

Retention Time [min.]	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	0.0	0.0	18.40833	18.58333	0.0	21.4	0.0	30.03333	33.75833	37.35833

Retention Time Residual [min.]	1	2	3	4	5	6	7	8	9	10
RT	15.077358	16.225707	17.370213	18.584032	21.263348	22.92186	24.547974	30.031305	33.762033	37.355372
GPC chromatogram	15.077358	16.225707	1.0381169999999997	0.0007020000000004245	21.263348	1.5218600000000002	24.547974	0.0020249999999997215	0.003703000000001566	0.002957999999999572
